### DISCUSSION TITLE
DT postgress Data Backup

### DISCUSSION THREAD

**Discussion by VASAVI512** [DependencyTrack/dependency-track#1882] [General]
Hello,

I am looking for DT postgress data backup process through dockers. Is there any document or how to see and check the data backup and restore.

Regards,
Vasavi

**Comment by nscuro** [comment]
Hi @VASAVI512, we don't have such a document right now, but it's something I have on my list of things to add docs for.

For now, for your specific question:
* Database backup & restore are not DT specific, just use whatever is best practice for the RDBMS you're using
* For the DT API server, the data directory should be backed up, more precisely the keys directory in it. When in doubt, just backup the whole data  directory. https://docs.dependencytrack.org/getting-started/data-directory/

**Reply by AsapHogFtw** [reply]
Hello, @nscuro - I am running Dependencytrack on a k8s cluster and wanted to know if it is important to backup the data dir if it is connecting to an external database. I am not able to use persistent storage on a k8s cluster, so I was think the backups might not be needed. We are already taking backups of the datbase.

Do I need to backup the data dir if using k8s and using an external DB (which is backed up) for DT to connect to?

**Comment by VASAVI512** [comment]
Ok. will use our RDMS.
And i am not finding application.properties files in my local machine where i have installed DT through docker. how to check about this file.
